### DISCUSSION TITLE
Support FN Key as Modifier

### DISCUSSION THREAD

**Discussion by mmmichl** [rxhanson/Rectangle#288] [Ideas]
Add support for the `fn` key as a modifier in addition to `ctrl`, `option` and `command`.

This would clear up the area of conflicts with other keyboard shortcuts! I saw the support in the following project, if this might help the implementation: https://github.com/finestructure/Hummingbird

**Comment by rxhanson** [comment]
This is actually a limitation imposed by the shortcut recorder, [MASShortcut](https://github.com/shpakovski/MASShortcut). I don't know the details, but I think there is a reason that the `fn` key isn't supported there. Note that `shift` is a supported modifier. If this does get implemented in MASShortcut, then I can pull the latest MASShortcut into Rectangle.

**Reply by smammy** [reply]
I just noticed that the MASShortcut repo was archived on March 5th of this year. Are you using any particular fork now?

**Reply by rxhanson** [reply]
Yes, I forked it a while ago but haven't made any modifications for Rectangle:
https://github.com/rxhanson/MASShortcut

The nice thing with owning it is that I can promptly roll in any changes anyone wants to make.

I have a private fork that I use for Rectangle Pro and Multitouch where I include a few more tweaks to it for recording only modifier keys, etc. 

It's actually a macOS restriction that prevents fn from being used in shortcuts.

**Reply by smammy** [reply]
Got it; thanks. I wonder how BetterTouchTool manages to support fn in shortcuts.

**Reply by rxhanson** [reply]
It likely listens to your keystrokes rather than registering shortcuts with the OS (this is something I prefer not to do with Rectangle/Pro). If you think of it that way, it's less of a macOS restriction and more of a convention that I prefer to follow.

**Reply by smammy** [reply]
No yeah, I'm totally with you on this one. BTT does a bunch of wild stuff that I would never do! 🤣

**Comment by yochem** [comment]
A workaround might be to install [Karabiner Elements](https://karabiner-elements.pqrs.org/) and remap the fn keys to something like cmd + ctrl + shift + num, so for example F1 would become cmd-ctrl-shift-1 etc. Then you can just use these in rectangle as shortcuts.
